packages/async-flow/test/async-flow-wake.test.js

```diff
@@ -0,0 +1,217 @@
+// The purpose of this test file is to demonstrate various wake behavior: if a
+// flow is either an eager waker or in a failed state it should be awoken in a
+// new incarnation, but otherwise should wait until a wake watcher is triggered.
+
+import { Fail } from '@endo/errors';
+import { testAsyncLife } from './prepare-test-env-ava.js';
+
+/**
+ * @import {Vow} from '@agoric/vow';
+ */
+
+/** @param {Pick<import('./prepare-test-env-ava.js').AsyncLifeTools, 'makeVowKit' | 'zone'>} tools */
+const makeTestKit = ({ zone, makeVowKit }) => {
+  const { resolver, vow } = zone.makeOnce('trigger', makeVowKit);
```

Comment:
I want to keep the option of adding arguments to `makeVowKit`, so please defend it against the `_key` argument supplied to the maker by `zone.makeOnce('key', maker)`.  I thought the TS types would warn about this, but I must've been wrong.

```suggestion
  const { resolver, vow } = zone.makeOnce('trigger', () => makeVowKit());
```